### ISSUE TITLE
How can I checkout the same git revision on a second build?

### ISSUE THREAD

**Issue by simmdan** [Microsoft/vsts-tasks#5137]
We have a system that allows us to have a VSTS build spawn one or more "child build(s)" which run in parallel to the parent build and then consolidate the results of the children so that we get an overall pass or failure on the main build.  We do this with a utility that calls out to the VSTS REST API to create the child builds and then later to monitor their status.  This part works fine and has been in use for some time.

Up until now, the child builds did not use our git-based source code and instead just operated on binaries shipped from the parent to the child.  Recently, however, we decided that we need the child build to checkout the same git revision as the parent.  So I added code that sends the revision of HEAD from the parent (output of `git rev-parse HEAD`) to the child and then a step on the child which does `git checkout` with that revision.  This works fine for cases where a build is queued manually from any branch or a specific commit ID, but when a build is triggered automatically as part of CI for a pull request, the parent build runs on the result of merging the branch from the PR with our master branch, and the revision returned by `git rev-parse HEAD` apparently is unavailable on the child build.

Looking at the output of the Get Sources step, I see that the fetch command which VSTS uses in this case looks something like:

`git` &lt;misc flags&gt; `--depth=1 origin +refs/heads/*:refs/remotes/origin/* +refs/pull/56503/merge:refs/remotes/pull/56503/merge`

The question for me is: where does it get the number 56503 or really that path in general.  It looks like that ref contains the revision I want, so if I could communicate it to the child build I would be golden, but when I'm inside a build task I don't know how to get it.  Is it a predefined variable? Or is there some code I can use to retrieve it?

**Comment by bryanmacfarlane** [comment]
move this to the vsts-agent repo.  That's where the get sources logic is.

we also have a feature about to come in preview which is multi phase (and parallel) support in build which may solve what you want as a first class feature.

**Comment by simmdan** [comment]
New issue in the vsts-agent repo is [here](https://github.com/Microsoft/vsts-agent/issues/1169).  Thx.
